File: src/js/sul/constants/hncom1.js
const Constants = {

	RemoteCommand: {

		origin: {

			HUB: 0,
			RCON: 2,
			EXTERNAL_CONSOLE: 1,

		},

	},

	Remove: {

		reason: {

			LEFT: 0,
			KICKED: 2,
			TIMED_OUT: 1,
			TRANSFERRED: 3,

		},

	},

	Add: {

		reason: {

			FORCIBLY_TRANSFERRED: 2,
			TRANSFERRED: 1,
			FIRST_JOIN: 0,

		},

	},

	Nodes: {

		type: {

			REMOVE: 1,
			ADD: 0,

		},

	},

	UpdateList: {

		list: {

			WHITELIST: 0,
			BLACKLIST: 1,

		},

		action: {

			REMOVE: 1,
			ADD: 0,

		},

	},

}

File: src/java/sul/constants/Hncom1.java
package sul.constants;

final class Hncom1 {

	private Hncom1() {}

	public final static class RemoteCommand {

		public final static class origin {

			public final static int HUB = 0;
			public final static int RCON = 2;
			public final static int EXTERNAL_CONSOLE = 1;

		}

	}

	public final static class Remove {

		public final static class reason {

			public final static int LEFT = 0;
			public final static int KICKED = 2;
			public final static int TIMED_OUT = 1;
			public final static int TRANSFERRED = 3;

		}

	}

	public final static class Add {

		public final static class reason {

			public final static int FORCIBLY_TRANSFERRED = 2;
			public final static int TRANSFERRED = 1;
			public final static int FIRST_JOIN = 0;

		}

	}

	public final static class Nodes {

		public final static class type {

			public final static int REMOVE = 1;
			public final static int ADD = 0;

		}

	}

	public final static class UpdateList {

		public final static class list {

			public final static int WHITELIST = 0;
			public final static int BLACKLIST = 1;

		}

		public final static class action {

			public final static int REMOVE = 1;
			public final static int ADD = 0;

		}

	}

}
